from fastapi import APIRouter, HTTPException, Depends
from sqlalchemy.orm import Session

from models import User, Task
from database import get_db

router = APIRouter()

@router.get("/user_id/{user_id}/tasks")
def tasks_by_user_id(user_id: int, db: Session = Depends(get_db)):
    tasks = db.query(Task).filter(Task.user_id == user_id).all()
    return tasks

@router.delete("/delete/{user_id}")
def delete_user(user_id: int, db: Session = Depends(get_db)):
    user = db.query(User).filter(User.id == user_id).first()
    if user is None:
        raise HTTPException(status_code=404, detail="User not found")

    # Удаляем все задачи пользователя
    db.query(Task).filter(Task.user_id == user_id).delete()
    db.delete(user)
    db.commit()

    return {'status_code': 200, 'transaction': 'User and related tasks deleted'}

# Удаляем User с id=1
# delete_user(user_id=1)  # Используйте ваш способ удаления пользователя